json-schema-provider/src/main/java/io/confluent/kafka/schemaregistry/json/JsonSchemaUtils.java

```diff
@@ -278,4 +280,80 @@ public static byte[] toJson(Object value) throws IOException {
     String jsonString = out.toString();
     return jsonString.getBytes(StandardCharsets.UTF_8);
   }
-}
\ No newline at end of file
+
+  public static JsonNode findMatchingField(JsonNode node, String path) {
+    String [] identifiers = path.split("\\.");
+    if (identifiers.length < 2) {
+      throw new IllegalArgumentException(
+        String.format("Given field path '%s' has less than two components.", path));
+    }
```

Comment:
I think this check is not application to JSON Schema and can be removed.  You might want to add a test to verify